### PR TITLE
Bugfix 6001/tN fix for external edit invalidation and loading selections

### PR DESCRIPTION
**Pull request by PhotoNomad0**
#### Describe what your pull request addresses (Please include relevant issue numbers):
- fix validateSelections() to support tN tool.
- fix load check data to support quote as an array.
- added updating of tool check data when checking resources are updated.

#### Please include detailed Test instructions for your pull request:
- test using steps and project in https://github.com/unfoldingWord-dev/translationCore/issues/6001

#### Standard Test Instructions for PR Review Process:

- [ ] Double check unit tests that have been written
- [ ] Check for documentation for code changes
- [ ] Check that there are not inadvertent commits to tC Apps when reviewing a tC Core PR
- [ ] Checkout the branch locally and ensure that app runs as expected
  - [ ] Ensure tests pass
  - [ ] Open and watch the console for errors
  - [ ] Make sure all actions perform as expected
  - [ ] Import and Load a new Project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Switch project to an existing project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Next time reverse the order of importing after loading an existing project
- [ ] Reviewer should double check the DoD in the ISSUE, including the “spirit” of the story
- [ ] Ask Ben or Koz if you have any concerns about implementation (especially UI related)

---
This change is [Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6155)

### CODE REVIEW

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (edfb7a9)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-linux-x64-1.2.0-edfb7a9.deb) is ready for download.
[macOS build 1.2.0 (edfb7a9)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-macos-x64-1.2.0-edfb7a9.dmg) is ready for download.
[Windows x64 build 1.2.0 (edfb7a9)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-win-x64-1.2.0-edfb7a9.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (edfb7a9)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-win-x32-1.2.0-edfb7a9.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (a0a1e8b)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-linux-x64-1.2.0-a0a1e8b.deb) is ready for download.
[macOS build 1.2.0 (a0a1e8b)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-macos-x64-1.2.0-a0a1e8b.dmg) is ready for download.
[Windows x64 build 1.2.0 (a0a1e8b)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-win-x64-1.2.0-a0a1e8b.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (a0a1e8b)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-win-x32-1.2.0-a0a1e8b.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (3e5d92b)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-linux-x64-1.2.0-3e5d92b.deb) is ready for download.
[macOS build 1.2.0 (3e5d92b)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-macos-x64-1.2.0-3e5d92b.dmg) is ready for download.
[Windows x64 build 1.2.0 (3e5d92b)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-win-x64-1.2.0-3e5d92b.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (3e5d92b)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-win-x32-1.2.0-3e5d92b.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (d700ebb)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-linux-x64-1.2.0-d700ebb.deb) is ready for download.
[macOS build 1.2.0 (d700ebb)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-macos-x64-1.2.0-d700ebb.dmg) is ready for download.
[Windows x64 build 1.2.0 (d700ebb)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-win-x64-1.2.0-d700ebb.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (d700ebb)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-win-x32-1.2.0-d700ebb.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (2b4a381)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-linux-x64-1.2.0-2b4a381.deb) is ready for download.
[macOS build 1.2.0 (2b4a381)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-macos-x64-1.2.0-2b4a381.dmg) is ready for download.
[Windows x64 build 1.2.0 (2b4a381)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-win-x64-1.2.0-2b4a381.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (2b4a381)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-win-x32-1.2.0-2b4a381.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (ee7d7e7)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-linux-x64-1.2.0-ee7d7e7.deb) is ready for download.
[macOS build 1.2.0 (ee7d7e7)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-macos-x64-1.2.0-ee7d7e7.dmg) is ready for download.
[Windows x64 build 1.2.0 (ee7d7e7)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-win-x64-1.2.0-ee7d7e7.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (ee7d7e7)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-win-x32-1.2.0-ee7d7e7.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (75f6f26)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-linux-x64-1.2.0-75f6f26.deb) is ready for download.
[macOS build 1.2.0 (75f6f26)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-macos-x64-1.2.0-75f6f26.dmg) is ready for download.
[Windows x64 build 1.2.0 (75f6f26)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-win-x64-1.2.0-75f6f26.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (75f6f26)](https://tc-builds.door43.org/build/bugfix-mcleanb-6001/translationCore-win-x32-1.2.0-75f6f26.setup.exe) is ready for download.

File: __tests__/fixtures/project/checkingResources/en_tit_checking/.apps/translationCore/checkData/selections/tit/1/4/2019-04-24T16_03_28.617Z.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "contextId": {
+    "reference": {
+      "bookId": "tit",
+      "chapter": 1,
+      "verse": 4
+    },
+    "tool": "translationWords",
+    "groupId": "godthefather",
+    "quote": "Θεοῦ Πατρὸς",
```
Comment (mannycolon) [inline __tests__/fixtures/project/checkingResources/en_tit_checking/.apps/translationCore/checkData/selections/tit/1/4/2019-04-24T16_03_28.617Z.json:10]: why is this quote still a string?

Comment (mannycolon) [reply __tests__/fixtures/project/checkingResources/en_tit_checking/.apps/translationCore/checkData/selections/tit/1/4/2019-04-24T16_03_28.617Z.json:10]: or was this done on purpose?

**Review by mannycolon — CHANGES_REQUESTED** [review]

**Review by mannycolon — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6155)* status: 1 of 369 files reviewed, 2 unresolved discussions (waiting on @PhotoNomad0)

___
*[package.json, line 130 at r6](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6155#-LhkpSFj0G3y69mNZhoZ:-LhkpSFkDDSXP18kjeaG:b956y6d) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/75f6f26f735369eb6754e7873d71b5aa4a6a1dd6/package.json#L130)):*
> ```JSON
>     "babel-preset-es2015": "6.24.1",
>     "babel-preset-react": "6.24.1",
>     "checking-tool-wrapper": "0.4.16",
> ```

Latest version is 0.4.17 merge w/ develop

**Review by mannycolon — COMMENTED** [review]
> Reviewed 1 of 3 files at r1, 1 of 2 files at r2, 1 of 2 files at r3, 364 of 365 files at r5, 1 of 1 files at r6.
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6155)* status: all files reviewed, 2 unresolved discussions (waiting on @PhotoNomad0)

**Review by RoyalSix — CHANGES_REQUESTED** [review]
> Reviewed 1 of 3 files at r1, 20 of 365 files at r5.
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6155)* status: all files reviewed, 3 unresolved discussions (waiting on @PhotoNomad0)

___
*[src/js/actions/SelectionsActions.js, line 233 at r6](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6155#-Lhkt9c15WOvsT6EnC3u:-Lhkt9c2-3hzbtZ8UR4S:bhvzua5) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/75f6f26f735369eb6754e7873d71b5aa4a6a1dd6/src/js/actions/SelectionsActions.js#L233)):*
> ```JavaScript
>       dispatch(validateAllSelectionsForVerse(targetVerse, results_, true, contextId, false, actionsBatch));
>       selectionInvalidated = selectionInvalidated || results_.selectionsChanged; // if new selections invalidated
>       selectionInvalidated = validateSelectionsForUnloadedTools(projectSaveLocation, bibleId, chapter, verse, state, targetVerse, selectionInvalidated, [getSelectedToolName(state)]);
> ```

Looks like the last parameter is not being used by the function that's being called `validateSelectionsForUnloadedTools`

File: __tests__/fixtures/project/checkingResources/en_tit_checking/.apps/translationCore/checkData/selections/tit/1/4/2019-04-24T16_03_28.617Z.json (hunk 1 above)
Comment (PhotoNomad0) [reply __tests__/fixtures/project/checkingResources/en_tit_checking/.apps/translationCore/checkData/selections/tit/1/4/2019-04-24T16_03_28.617Z.json:10]: Yes, that is the data before doing the resource updates.

**Review by PhotoNomad0 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6155)* status: all files reviewed, 3 unresolved discussions (waiting on @mannycolon and @RoyalSix)

___
*[package.json, line 130 at r6](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6155#-LhkpSFj0G3y69mNZhoZ:-LhkuTeTElcrR-SPkyHn:bmteo9m) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/75f6f26f735369eb6754e7873d71b5aa4a6a1dd6/package.json#L130)):*
<details><summary><i>Previously, mannycolon (Manny Colon) wrote…</i></summary><blockquote>

Latest version is 0.4.17 merge w/ develop
</blockquote></details>

That is not mine, that's Jay's from today on a different issue.
___
*[src/js/actions/SelectionsActions.js, line 233 at r6](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6155#-Lhkt9c15WOvsT6EnC3u:-LhkvTDI--sul47sRuSO:b53eap9) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/75f6f26f735369eb6754e7873d71b5aa4a6a1dd6/src/js/actions/SelectionsActions.js#L233)):*
<details><summary><i>Previously, RoyalSix (Sterling (Jay) Scott) wrote…</i></summary><blockquote>

Looks like the last parameter is not being used by the function that's being called `validateSelectionsForUnloadedTools`
</blockquote></details>

good catch, was passing as a parameter and then realized it wasn't necessary.

**Review by mannycolon — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6155)* status: all files reviewed, 1 unresolved discussion (waiting on @RoyalSix)

**Pull request merged**